internal/curator/durable/state/state.go

```diff
@@ -101,18 +101,23 @@ func Open(path string) *State {
 	if err != nil {
 		log.Fatalf("Failed to start a transaction: %v", err)
 	}
-	if _, err := tx.CreateBucketIfNotExists(partitionBucket); err != nil {
-		log.Fatalf("Failed to create partition bucket: %v", err)
-	}
-	if _, err := tx.CreateBucketIfNotExists(blobBucket); err != nil {
-		log.Fatalf("Failed to create blob bucket: %v", err)
-	}
-	if _, err := tx.CreateBucketIfNotExists(rschunkBucket); err != nil {
-		log.Fatalf("Failed to create rschunk bucket: %v", err)
+
+	// Check if we're starting with an empty db.
+	isNewDb := tx.Bucket(metaBucket) == nil
+
+	for _, bucket := range [][]byte{
+		partitionBucket, blobBucket, rschunkBucket, metaBucket,
+	} {
+		if _, err := tx.CreateBucketIfNotExists(bucket); err != nil {
+			log.Fatalf("Failed to create bucket %q: %v", bucket, err)
+		}
 	}
-	if _, err := tx.CreateBucketIfNotExists(metaBucket); err != nil {
-		log.Fatalf("Failed to create id bucket: %v", err)
+
+	if isNewDb {
+		// If we're starting from scratch, we can use the tsid cache without explicit commands.
```

Comment:
Is the metaBucket a generic bucket that can contain whatever, including the tsid cache?

Reply:
Yeah, it's just a place to put all the singletons. They could be in their own buckets, but it seems simpler to keep them in one.